clang/include/clang/AST/DeclCXX.h

```diff
@@ -1960,24 +1960,38 @@ class ExplicitSpecifier {
 class CXXDeductionGuideDecl : public FunctionDecl {
   void anchor() override;
 
+public:
+  enum class SourceKind {
```

Comment:
Can we get a better name / comment for this enum?

Reply:
Done, renamed to `SourceDeductionGuideKind` to align with the class member and added a comment

Reply:
@cor3ntin, if you have any ideas for a better name, could you suggest one?

I think adding a few comments to the enum should suffice. Personally, I like `SourceKind` — it's short, clear, and works well as `CXXDeductionGuideDecl::SourceKind::XXX`.

On the other hand, `SourceDeductionGuideKind` feels a bit lengthy and redundant with "DeductionGuide" already in the context.